### ISSUE TITLE
TestsCore

### ISSUE THREAD

**Issue by Jimilian** [SmallButTeam/LandOfDKL_project#33]
Вам во всех тестах нужна база?

**Comment by abondar** [comment]
Нет, не во всех - но в нескольких - какое решение лучше? Каждый раз создавать новый HibernateUtil в каждом классе использующем базу слишком долго.

**Comment by Jimilian** [comment]
> какое решение лучше?
> Можно сделать абстрактный DatabaseTests, реализовывать который будут только те тесты, что работают с базой.

---

> Каждый раз создавать новый HibernateUtil в каждом классе использующем базу слишком долго.
> Вы именно это сейчас и делаете.

**Comment by abondar** [comment]
"Можно сделать абстрактный DatabaseTests, реализовывать который будут только те тесты, что работают с базой."
"Вы именно это сейчас и делаете."
У на сейчас соединение с базой создаётся в статическом методе и происходит до всех тестов, все кому надо просто используют уже готовую Фабрику сессий. Если создать абстрактный класс, то в каждой реализации будет создаваться соединение с базой данных, а по нашим подсчетам соединение с базой происходит за 3 секунды. Таким образом имея 3+ классов использующих базу тратим как минимум на 6 секунд больше на тесты. Если я конечно правильно понял, то, что вы имели ввиду.

UPDATE только что проверил - да я похоже что-то не так понимал в том как работает "@BeforeClass", он действительно заново создает утил - но ведь при том раскладе, если будет реализация абстрактного метода, то он тоже каждый раз его будет создавать, разве нет?
